<launch>
  <node pkg="mil_passive_sonar"
        type="passive_sonar_driver.py"
        name="passive_sonar_driver"
        output="screen">
    <!-- units in millimeters -->
    
    <rosparam>
      method : "LS"
      c : 1.484
      receiver_locations: <!-- SubjuGator 8 (2017) configuration -->
        - x : -25.4
          y :     0
          z :  25.4
        - x :  25.4
          y :     0
          z :     0
        - x :     0
          y : -25.4
          z :     0
      port : "/dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A901AN8P-if00-port0"
      baud : 115200
      tx_request_code : 'B'
      tx_start_code : "\xBB"
      target_freq : 35000
      scalar_size : 5
      signal_size : 300
      signal_bias : 32767.0
      sampling_freq : 486618
      upsample_factor : 20
    </rosparam>
  </node>
</launch>

